curry_bot.py

```diff
@@ -97,70 +88,7 @@ async def bingo(ctx):
-        if args[0].startswith('preset'):
-            print_links = False
-            await ctx.send('\n'.join(curry_format("Preset: {}, Description: {}", preset, description) for preset, description in RANDO_PRESETS.items()))
-        else:
-            updated = False
-            preset = get_matching_preset(args[0])
-            if preset:
-                quantity = SEEDS_TO_GENERATE
-                if len(args) >= 2:
-                    if args[1].isdigit():
-                        user_quantity = int(args[1])
-                        if user_quantity <= 0 or user_quantity > MAX_SEEDS:
-                            await ctx.send(curry_format("Number of seeds is limited to {}.", MAX_SEEDS))
-                            quantity = 0
-                        else:
-                            quantity = user_quantity
-                    else:
-                        await ctx.send(curry_format("I don't know how to generate {} seeds. Type '!help rando' for help.", args[1]))
-                        quantity = 0
-                if quantity > 0:
-                    await ctx.send(curry_format("Generating {} {} seeds...", quantity, RANDO_PRESETS[preset]))
-                    seed_base = time.time() * 1000
-                    seed_floor = math.floor(seed_base)
-                    # Use the nanosecond portion of the time as a pseudo-random offset, plus a constant in case that happens to be 0
-                    offset = math.floor((seed_base - seed_floor) * 1000) + 50
-                    RANDO_LINKS = [make_seed(preset, seed_floor - offset * i) for i in range(quantity)]
-                    updated = True
-                else:
-                    print_links = False
-            elif args[0].startswith(RANDO_BASE):
-                RANDO_LINKS = args
-                updated = True
-            else:
-                await ctx.send(curry_message("That doesn't look like a seed or preset. Curry."))
-                print_links = False
-            await ctx.send(curry_format("Rando seed links {}updated", "" if updated else "NOT "))
-    if print_links:
-        await ctx.send(curry_message("Current rando seed links:"))
-        await ctx.send('\n'.join(curry_format("Seed {}: <{}>", i+1, link) for i, link in enumerate(RANDO_LINKS)))
-
-
-def get_matching_preset(param):
-    preset = NULL_PRESET
-    # This is necessary to handle cases where one preset contains the name of another preset
-    if param in RANDO_PRESETS:
-        preset = param
-    else:
-        # Search for partial matches to allow e.g. 'star' to work for 'starsTournament'
-        matches = 0
-        for key in RANDO_PRESETS.keys():
-            if key.startswith(param):
-                preset = key
-                matches += 1
-        if matches != 1:
-            preset = NULL_PRESET
-    return preset
-
-
-def make_seed(preset, seed):
-    return "{}?P:{},,{}".format(RANDO_BASE, preset, seed)
+    await RandoCommandHandler(ctx, args).handle()
```

Comment:
I'm attempting to keep to the pattern that 'curry_bot.py' handles *all* Discord messaging stuff, not any of the imported modules (perhaps I've violated that rule in a couple of places, but will try to correct that). The advantage of this would be that we keep all the Discord IO in one place (this file, or a dedicated Discord wrapper module in the future), so that CurryBot isn't saying anything we don't expect. Also, it helps to keep the modules bot-agostic, i.e. they don't care about the nature of the bot, and don't use things like Curry emotes.

To follow this pattern, maybe it is best for RandoCommandHandler to return an iterable of strings, and then here in curry_bot we take that list and create the messages.

Reply:
Fixed in the latest commit.